### DISCUSSION TITLE
Repair orders for repair shop

### DISCUSSION THREAD

**Discussion by tedd-wq** [inventree/InvenTree#7694] [Ideas]
I run a repair service. From what I can see, Inventree is not able to create a repair order, and to track parts and a description of repair work done to a customer's unit. Is this a feature that would be likely implemented?

**Comment by SchrodingersGat** [comment]
@tedd-wq potential something we could add - would you be willing to assist with implementation or contribute / fund development efforts?

**Reply by cameleonnzl** [reply]
Hiya. replying to this very old thread :)

I'm maybe able to assist with implementing, and could possibly help with funding or development depending on what would be required. I restore vintage mac's and also assemble small electronic kits based off BOM's - which led me to Inventree in the first place.
I've stopped using my current repair web portal though and am back using a spreadsheet and an external invoicing partner (Xero) as it was too expensive for what I was getting. I installed Inventree today (just as you released 1.0.1 it seems).

**Comment by VMT-Solutions** [comment]
i would also be willing to provide funding for this feature request

**Comment by ataylor-rp** [comment]
As would I.

**Comment by SchrodingersGat** [comment]
I have created a new issue to track this - https://github.com/inventree/InvenTree/issues/12064

**Comment by Sajad100k** [comment]
o define the repair workflow properly and see how it can fit into InvenTree, I need to clarify a few things about how you currently handle repairs. The goal is to make sure the system matches your real process and doesn’t become more complicated than it needs to be.

When a customer brings in a device, what normally happens first? Do you need a repair order created immediately, or is there usually an inspection or diagnostic step before that? Also, if a device turns out to be beyond repair, how should that be handled in the system?

I also need to understand how you want customer devices to be tracked. Does each unit need its own unique record based on a serial number, IMEI, or internal tag, or is it enough to store general model information? It would also help to know whether you want to see repair history for the same device over time.

On the inventory side, when a part is used during a repair, should it be deducted directly from stock? Or would you prefer a process where parts are first reserved and only consumed once the repair is completed? Along with that, do you need the system to record non-part charges such as labor, diagnostic fees, or shipping?

It would also be useful to define the repair statuses you want to work with. For example, do you need stages such as received, under inspection, waiting for parts, in progress, ready for pickup, and completed? And when work is performed, is a simple repair note enough, or do you also want to track technician time and detailed work logs?

Please also let me know what kind of documentation you need around each repair. Should the system support intake notes, customer signatures, attached photos, or final repair summaries? And from an access point of view, do different staff members need different permissions, such as front desk staff creating jobs while technicians update work performed?

Finally, for the first version, what would you consider the minimum set of features needed to make this usable in day-to-day operations? I’d prefer to define a practical first phase and then leave anything less essential for later.

Once I have your answers, I can turn this into a clear feature scope covering the data model, UI, and API requirements.
